In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

plt.style.use("ggplot")

In [2]:
file = r"../Dataset/Cleaned_Academic_Probation_Project.xlsx"

df = pd.read_excel(
    file,
    sheet_name="Master_Education_Data (2)"
)

df.head()

,Student_ID,Student_Name,Gender,Department,Semester,Attendance_Percentage,GPA,Assignment_Score,Quiz_Score,Internal_Marks,LMS_Login_Count,Study_Hours_Per_Week,Backlog_Count,Last_Login_Date,Admission_Year,Probation_Status
0,S0001,Ajay Das,Male,AI&DS,8,57.0,6.67,76,38,95,41,35,2,2025-07-28 00:00:00,2023,Yes
1,S0002,Pavithra Murthy,Female,IT,1,83.0,7.08,91,62,41,234,29,1,2026-05-27 00:00:00,2022,No
2,S0003,Sandhya Das,Female,EEE,5,72.0,6.20,83,87,55,3,3,2,2026-04-19 00:00:00,2026,No
3,S0004,Rekha Balakrishnan,Female,AI&DS,3,67.0,6.08,55,91,89,3,17,2,2026-02-03 00:00:00,2024,No
4,S0005,Balaji Gupta,Male,EEE,5,44.0,2.31,84,62,80,166,2,3,2026-01-03 00:00:00,2024,Yes


In [3]:
total_students = len(df)

avg_gpa = df["GPA"].mean()

avg_attendance = df["Attendance_Percentage"].mean()

avg_assignment = df["Assignment_Score"].mean()

avg_quiz = df["Quiz_Score"].mean()

avg_internal = df["Internal_Marks"].mean()

avg_lms = df["LMS_Login_Count"].mean()

avg_study = df["Study_Hours_Per_Week"].mean()

avg_backlog = df["Backlog_Count"].mean()

probation_yes = (df["Probation_Status"] == "Yes").sum()

probation_no = (df["Probation_Status"] == "No").sum()

In [4]:
print("="*50)

print("STUDENT PERFORMANCE KPIs")

print("="*50)

print(f"Total Students            : {total_students}")

print(f"Average GPA               : {avg_gpa:.2f}")

print(f"Average Attendance        : {avg_attendance:.2f}%")

print(f"Average Assignment Score  : {avg_assignment:.2f}")

print(f"Average Quiz Score        : {avg_quiz:.2f}")

print(f"Average Internal Marks    : {avg_internal:.2f}")

print(f"Average Study Hours       : {avg_study:.2f}")

print(f"Average LMS Logins        : {avg_lms:.2f}")

print(f"Average Backlogs          : {avg_backlog:.2f}")

print(f"Students on Probation     : {probation_yes}")

print(f"Students not on Probation : {probation_no}")

STUDENT PERFORMANCE KPIs
Total Students            : 10000
Average GPA               : 5.71
Average Attendance        : 69.99%
Average Assignment Score  : 69.58
Average Quiz Score        : 69.73
Average Internal Marks    : 69.65
Average Study Hours       : 20.17
Average LMS Logins        : 80.25
Average Backlogs          : 2.86
Students on Probation     : 7881
Students not on Probation : 2119


In [8]:
department_kpi = (
    df.groupby("Department")
      .agg(
          Average_GPA=("GPA","mean"),
          Average_Attendance=("Attendance_Percentage","mean"),
          Average_Study_Hours=("Study_Hours_Per_Week","mean"),
          Average_LMS_Login=("LMS_Login_Count","mean"),
          Students=("Student_ID","count")
      )
      .round(2)
)

department_kpi

,Average_GPA,Average_Attendance,Average_Study_Hours,Average_LMS_Login,Students
Department,,,,,
AI&DS,5.61,71.09,20.19,80.24,1409
CIVIL,5.76,69.24,20.19,80.58,1424
CSE,5.75,69.68,19.70,81.07,1423
ECE,5.74,69.84,20.10,80.34,1472
EEE,5.63,69.56,20.15,79.60,1451
IT,5.76,70.12,20.77,81.02,1399
MECH,5.68,70.42,20.11,78.88,1422


In [10]:
def risk_level(row):
    if (
        row["Attendance_Percentage"] < 50 or
        row["GPA"] < 5 or
        row["Backlog_Count"] >= 3
    ):
        return "Critical Risk"

    elif (
        row["Attendance_Percentage"] < 75 or
        row["GPA"] < 6.5
    ):
        return "At Risk"

    else:
        return "Normal"

df["Risk_Level"] = df.apply(risk_level, axis=1)

In [11]:
risk_summary = (
    df["Risk_Level"]
      .value_counts()
      .reset_index()
)

risk_summary.columns = ["Risk Level","Students"]

risk_summary

,Risk Level,Students
0,Critical Risk,7461
1,At Risk,1561
2,Normal,978


In [12]:
probation_summary = (
    df["Probation_Status"]
      .value_counts()
      .reset_index()
)

probation_summary.columns = ["Probation Status", "Students"]

probation_summary

,Probation Status,Students
0,Yes,7881
1,No,2119


In [13]:
department_kpi.to_csv(
    "Output_Charts/Department_KPI_Table.csv"
)

risk_summary.to_csv(
    "Output_Charts/Risk_Level_Summary.csv",
    index=False
)

probation_summary.to_csv(
    "Output_Charts/Probation_Summary.csv",
    index=False
)

# Insights

1. The dataset contains a total of 10,000 student records.

2. The average GPA of students is approximately 5.71.

3. The average attendance percentage is around 70%.

4. A significant number of students fall under the Critical Risk category, indicating the need for early academic intervention.

5. Student engagement, measured using LMS login activity and study hours, varies across departments.

6. Department-wise KPI analysis helps identify departments that require additional academic support.

7. The calculated KPIs provide a clear overview of student performance and can support data-driven academic decisions.

# Recommendations

1. Monitor students with low GPA and attendance on a regular basis.

2. Provide mentoring and counseling for students classified as Critical Risk.

3. Encourage students to improve LMS engagement through assignments and quizzes.

4. Conduct regular academic performance reviews using KPI dashboards.

5. Implement early intervention programs for students showing signs of academic decline.

6. Use the Power BI dashboard for continuous monitoring and decision-making.

In [14]:
print("="*60)
print("DAY 6 - KPI MONITORING DASHBOARD COMPLETED")
print("="*60)

print(f"Total Students           : {total_students}")
print(f"Average GPA              : {avg_gpa:.2f}")
print(f"Average Attendance       : {avg_attendance:.2f}%")
print(f"Students on Probation    : {probation_yes}")
print(f"Students not on Probation: {probation_no}")

print("\nKPI Tables exported successfully.")
print("Ready for Power BI Dashboard Development (Day 7)")

DAY 6 - KPI MONITORING DASHBOARD COMPLETED
Total Students           : 10000
Average GPA              : 5.71
Average Attendance       : 69.99%
Students on Probation    : 7881
Students not on Probation: 2119

KPI Tables exported successfully.
Ready for Power BI Dashboard Development (Day 7)
